GitCommands/Git/GitModule.cs

```diff
@@ -3188,111 +3188,219 @@ public GitBlame Blame(string filename, string from, Encoding encoding)
+
+            // Where one commit is responsible for multiple contiguous lines, they appear as follows:
+
+            // e3268019c66da7534414e9562ececdee5d455b1b 11 13 9
+            //         <table>
+            // e3268019c66da7534414e9562ececdee5d455b1b 12 14
+            //           <tr>
+            // e3268019c66da7534414e9562ececdee5d455b1b 13 15
+            //             <th>&nbsp;</th>
+            // e3268019c66da7534414e9562ececdee5d455b1b 14 16
+            //             <th>Windows</th>
+            // e3268019c66da7534414e9562ececdee5d455b1b 15 17
+            //             <th>Linux/Mac</th>
+            // e3268019c66da7534414e9562ececdee5d455b1b 16 18
+            //           </tr>
+            // e3268019c66da7534414e9562ececdee5d455b1b 17 19
+            //           <tr>
+            // e3268019c66da7534414e9562ececdee5d455b1b 18 20
+            //             <td>Requirement</td>
+            // e3268019c66da7534414e9562ececdee5d455b1b 19 21
+
+            // On the first line here, there are three trailing numbers. The third is the number of
+            // lines from the one commit.
+
+            // There are three 'chunks' here. The first includes a commit header.
+            // The second and third refer back to a header that would have been presented earlier.
+            // We see the content of the chunks, where the first is a markdown header, the second
+            // is a blank line, and third is an introductory paragraph about the project.
+
+            var commitByObjectId = new Dictionary<string, GitBlameCommit>();
+            var lines = new List<GitBlameLine>(capacity: 256);
+
+            var headerRegex = new Regex(@"^(?<objectid>[0-9a-f]{40}) (?<origlinenum>\d+) (?<finallinenum>\d+)", RegexOptions.Compiled);
+
+            bool hasCommitHeader;
+            string objectId;
+            int finalLineNumber;
+            int originLineNumber;
+            string author;
+            string authorMail;
+            DateTime authorTime;
+            string authorTimeZone;
+            string committer;
+            string committerMail;
+            DateTime committerTime;
+            string committerTimeZone;
+            string summary;
+            string filename;
+
+            void Reset()
+            {
+                hasCommitHeader = false;
+                objectId = null;
+                finalLineNumber = -1;
+                originLineNumber = -1;
+                author = null;
+                authorMail = null;
+                authorTime = DateTime.MinValue;
+                authorTimeZone = null;
+                committer = null;
+                committerMail = null;
+                committerTime = DateTime.MinValue;
+                committerTimeZone = null;
+                summary = null;
+                filename = null;
+            }
+
+            Reset();
+
+            foreach (var line in output.Split('\n').Select(l => l.TrimEnd('\r')))
```

Comment:
There was lots of trimming everywhere to deal with training `\r` after splitting on `\n`. This sorts it all out in one place.